forte/data/caster.py

```diff
@@ -78,3 +78,43 @@ def input_pack_type():
     @staticmethod
     def output_pack_type():
         return MultiPack
+
+
+class DataPackBoxer(Caster[MultiPack, DataPack]):
+    """
+    This class creates a DataPack from a MultiPack, this DataPack is the only
+    contain of the original MultiPack, indexed by the :attr:`pack_name`.
+    """
+
+    def cast(self, pack: MultiPack) -> DataPack:
+        """
+        Auto-box the multi-pack into a data-pack by simple getting pack.
+
+        Args:
+            pack: The data pack to be boxed
+
+        Returns: An iterator that produces the boxed data-pack.
+
+        """
+        if pack.pack_name:
+            pack_name = (
+                pack.pack_name.strip("_multi")
+                if "_multi" in pack.pack_name
+                else pack.pack_name
+            )
+        else:
+            pack_name = None
```

Comment:
Doesn't quite make sense here, if `pack_name` is None, then what's the behavior of `get_pack`?

There are a couple of ways that I consider how this boxer would work:
1. allow using `pack_index` to take a pack, and we can make the default value to be 0
2. check whether `pack_index` and `pack_name` both exist and try to pick the preferred one
3. throw an exception during the initialize stage to check if we have valid `pack_name` and `pack_index`